### PR TITLE
Remove example-snap package

### PR DESCRIPTION
**Pull request by rekmarks**
Deletes the `example-snap` package which we should deprecate in favor of: https://github.com/MetaMask/template-snap.

### CODE REVIEW

File: packages/snaps-cli/scripts/createInitTemplate.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ async function createInitTemplate() {
     `${JSON.stringify(
       {
         html: html.toString(),
-        source: js.toString(),
+        source: js.toString().replace(/\r\n/gu, '\n'),
```
Comment (hmalik88) [inline packages/snaps-cli/scripts/createInitTemplate.js]: Do we need to have the check for new line if we're ultimately replacing with newline? Can it be just `.replace(/\r/gu, '\n')` ?

Comment (hmalik88) [reply packages/snaps-cli/scripts/createInitTemplate.js]: Nevermind! for some reason I thought it was or 😅

**Review by hmalik88 — DISMISSED** [review]

File: packages/snaps-cli/scripts/createInitTemplate.js (hunk 1 above)
Comment (rekmarks) [reply packages/snaps-cli/scripts/createInitTemplate.js]: Good question. Whenever I do regex manipulations of strings in this manner, I try to preclude undesired modifications to the greatest extent possible. This `.replace` call is meant to replace CRLF line breaks with LF line breaks and nothing else, so we try to do just that.

Comment (rekmarks) [reply packages/snaps-cli/scripts/createInitTemplate.js]: I added comments to explain this, and simplified the call.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 **/dist
+**/temp-*
```
Comment (rekmarks) [inline .gitignore:2]:
```suggestion
```

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**